src/abstract/TieredLiquidityDistributor.sol

```diff
@@ -29,494 +29,1135 @@ error InsufficientLiquidity(uint104 requestedLiquidity);
+  uint32 internal immutable ESTIMATED_PRIZES_PER_DRAW_FOR_11_TIERS;
+  uint32 internal immutable ESTIMATED_PRIZES_PER_DRAW_FOR_12_TIERS;
+  uint32 internal immutable ESTIMATED_PRIZES_PER_DRAW_FOR_13_TIERS;
+  uint32 internal immutable ESTIMATED_PRIZES_PER_DRAW_FOR_14_TIERS;
 
-    /// @notice Adjusts the number of tiers and distributes new liquidity
-    /// @param _nextNumberOfTiers The new number of tiers. Must be greater than minimum
-    /// @param _prizeTokenLiquidity The amount of fresh liquidity to distribute across the tiers and reserve
-    function _nextDraw(uint8 _nextNumberOfTiers, uint96 _prizeTokenLiquidity) internal {
-        if(_nextNumberOfTiers < MINIMUM_NUMBER_OF_TIERS) {
-            revert NumberOfTiersLessThanMinimum(_nextNumberOfTiers);
-        }
-
-        uint8 numTiers = numberOfTiers;
-        UD60x18 _prizeTokenPerShare = fromUD34x4toUD60x18(prizeTokenPerShare);
-        (uint16 completedDrawId, uint104 newReserve, UD60x18 newPrizeTokenPerShare) = _computeNewDistributions(numTiers, _nextNumberOfTiers, _prizeTokenPerShare, _prizeTokenLiquidity);
-
-        // need to redistribute to the canary tier and any new tiers (if expanding)
-        uint8 start;
-        uint8 end;
-        // if we are expanding, need to reset the canary tier and all of the new tiers
-        if (_nextNumberOfTiers > numTiers) {
-            start = numTiers - 1;
-            end = _nextNumberOfTiers;
-        } else { // just reset the canary tier
-            start = _nextNumberOfTiers - 1;
-            end = _nextNumberOfTiers;
-        }
-        for (uint8 i = start; i < end; i++) {
-            _tiers[i] = Tier({
-                drawId: completedDrawId,
-                prizeTokenPerShare: prizeTokenPerShare,
-                prizeSize: uint96(_computePrizeSize(i, _nextNumberOfTiers, _prizeTokenPerShare, newPrizeTokenPerShare))
-            });
-        }
-
-        prizeTokenPerShare = fromUD60x18toUD34x4(newPrizeTokenPerShare);
-        numberOfTiers = _nextNumberOfTiers;
-        lastCompletedDrawId = completedDrawId;
-        _reserve += newReserve;
+  UD60x18 internal immutable CANARY_PRIZE_COUNT_FOR_2_TIERS;
+  UD60x18 internal immutable CANARY_PRIZE_COUNT_FOR_3_TIERS;
+  UD60x18 internal immutable CANARY_PRIZE_COUNT_FOR_4_TIERS;
+  UD60x18 internal immutable CANARY_PRIZE_COUNT_FOR_5_TIERS;
+  UD60x18 internal immutable CANARY_PRIZE_COUNT_FOR_6_TIERS;
+  UD60x18 internal immutable CANARY_PRIZE_COUNT_FOR_7_TIERS;
+  UD60x18 internal immutable CANARY_PRIZE_COUNT_FOR_8_TIERS;
+  UD60x18 internal immutable CANARY_PRIZE_COUNT_FOR_9_TIERS;
+  UD60x18 internal immutable CANARY_PRIZE_COUNT_FOR_10_TIERS;
+  UD60x18 internal immutable CANARY_PRIZE_COUNT_FOR_11_TIERS;
+  UD60x18 internal immutable CANARY_PRIZE_COUNT_FOR_12_TIERS;
+  UD60x18 internal immutable CANARY_PRIZE_COUNT_FOR_13_TIERS;
+  UD60x18 internal immutable CANARY_PRIZE_COUNT_FOR_14_TIERS;
 
-    }
+  SD59x18 internal immutable TIER_ODDS_0_3;
```

Comment:
Tier odds will always be less than 0 <= 1, so you can use the 64-bit SD1x18.

This should save gas